components/webgpu/lib.rs

```diff
@@ -152,6 +166,41 @@ impl WGPU {
                         )
                     }
                 },
+                WebGPURequest::CreateBuffer(sender, device, id, descriptor) => {
+                    let global = &self.global;
+                    let _output =
+                        gfx_select!(id => global.device_create_buffer(device.0, &descriptor, id));
+                    let buffer = WebGPUBuffer(id);
+                    if let Err(e) = sender.send(buffer) {
+                        warn!(
+                            "Failed to send response to WebGPURequest::CreateBuffer ({})",
+                            e
+                        )
+                    }
+                },
+                WebGPURequest::CreateBufferMapped(sender, device, id, descriptor) => {
+                    let global = &self.global;
+                    let mut arr_buff: *mut u8 = std::ptr::null_mut();
+
+                    let _output = gfx_select!(id =>
+                        global.device_create_buffer_mapped(device.0, &descriptor, &mut arr_buff, id));
+                    let buffer = WebGPUBuffer(id);
+
+                    if let Err(e) = sender.send((buffer, arr_buff as usize)) {
```

Comment:
This sends a pointer over IPC, so this won't work with multiple processes. How do we deal with this? Do we need to copy the memory into a Vec and send that instead?

Reply:
Yes, you are right. I've copied the memory into a `Vec` as you mentioned.